chain/chain/src/store.rs

```diff
@@ -1137,12 +1137,10 @@ impl ChainAccessForFlatStorage for ChainStore {
     }
 
     fn get_block_hashes_at_height(&self, height: BlockHeight) -> HashSet<CryptoHash> {
-        self.get_all_block_hashes_by_height(height)
-            .unwrap()
-            .values()
-            .flatten()
-            .copied()
-            .collect::<HashSet<_>>()
+        match self.get_all_block_hashes_by_height(height) {
+            Ok(hashes) => hashes.values().flatten().copied().collect::<HashSet<_>>(),
+            Err(_) => Default::default(),
```

Comment:
Hm, this change I think warrants a comment: we are silently swallowing an error here. It also feels inconsistent that we have two methods here, one swallows errors, and one don't.


Not sure what's the right behavior here :)

Reply:
I'm also a bit confused because it is the same for `ChainStore` methods: errors after `get_all_block_hashes_by_height` calls are swallowed, and `get_block_header` don't.

I think that we need to fix `ChainStore::get_all_block_hashes_by_height` method: instead of `option_to_not_found` conversion, we should return an empty hashmap if element was not found. It is a normal behaviour, because some heights can be skipped.

It would take some effort. Let's file a separate issue for this?

Reply:
https://github.com/near/nearcore/issues/8057